cli/cmd/install.go

```diff
@@ -434,6 +436,10 @@ func (options *installOptions) validateAndBuild(stage string, flags *pflag.FlagS
 		return nil, nil, err
 	}
 
+	if err = validateEndpointSlicesFeature(options.enableEndpointSlices); err != nil {
+		return nil, nil, errors.New("--enableEndpointSlice enabled but resource non-existent in cluster")
```

Comment:
This is hiding the error returned by `validateEndpointSlicesFeature()`, which might not necessarily be that the resource is non-existent.

Reply:
Hi @alpeb  thanks for the comments! I haven't used helm much so the `charts/linkerd2/templates/_config.tpl` file escaped me! Will run some manual tests with helm installations to make sure everything works as intended 👍 

For `validateEndpointSlicesFeature()` I have changed the returned error by chaining `--enableEndpointSlice true not supported:` with whatever error `validateEndpointSlicesFeature` returns. I could also take another crack at `validateEndpointSlicesFeature()` and return some more customised errors instead of chaining them if you think that'd be a better approach.

Reply:
Thanks @Matei207 chaining errors like that works for me :+1: